examples-gallery/beginner/plot_wheel_on_bumpy_road.py

```diff
@@ -0,0 +1,425 @@
+               eom,
+               state_symbols,
+               num_nodes,
+               interval,
+               known_parameter_map=par_map,
+               instance_constraints=instance_constraints,
+               bounds=bounds,
+               time_symbol=t,
+               backend='numpy',
+    )
+
+    solution = np.loadtxt(fname)
+else:
+    # Iterate to find the solution. As the convergence is not easy, one has to
+    # start with a smooth road and then increase the roughness gradually.
+    # Before the bounds have to be tightended gardually
+    for i in range(5):
+        for j in range(5):
+            bounds[prevent_jump] = (-4.0+j, 4.0-j + 0.1)
+            bounds[steady_body] = (-4.0+j+0.85, 4.0-j+1.0)
+            if i < 5:
+                par_map[r3] = 0.1 + 0.0725*i
+                par_map[r4] = 0.0725*i
+
+            prob = Problem(obj,
+               obj_grad,
+               eom,
+               state_symbols,
+               num_nodes,
+               interval,
+               known_parameter_map=par_map,
+               instance_constraints=instance_constraints,
+               bounds=bounds,
+               time_symbol=t,
+            )
+
+            prob.add_option('max_iter', 3000)
+            if i == 0:
+                initial_guess = np.random.rand(prob.num_free)
+            else:
+                initial_guess = solution
+            for _ in range(3):
+                solution, info = prob.solve(initial_guess)
+                initial_guess = solution
+                print(info['status_msg'])
+                print('Objective value', info['obj_val'])
+#np.savetxt(fname, solution, fmt='%.12f')
+# %%
+# Print optimal values of the free parameters.
+print('Sequence of unknown parameters',
+               prob.collocator.unknown_parameters)
+print(f'optimal value of dampening constant c =                  ' +
+      f'{solution[-4]:.2f}')
+print(f'optimal value of spring constant k =                     ' +
+      f'{solution[-3]:.3f}')
+#print(f'optimal value of wheel spring constant k1 =              ' +
+#      f'{solution[-3]:.2f}')
+print(f'optimal value of nat.  length of the wheel spring l_GW = ' +
+      f'{solution[-2]:.2f}')
+
+# %%
+fig, axes = plt.subplots(11, 1, figsize=(7, 20), layout='tight')
+_ = prob.plot_trajectories(solution, axes=axes)
+# %%
+_ = prob.plot_constraint_violations(solution)
+# %%
+# Plot the road.
+r11, r22, r33, r44, r55 = [par_map[key] for key in [r1, r2, r3, r4, r5]]
+rough_surface_lam = sm.lambdify((x_car, r1, r2, r3, r4, r5),
+                                rough_surface(x_car), cse=True)
+XX = np.linspace(0, 10, 100)
+r11, r22, r33, r44, r55 = [par_map[key] for key in [r1, r2, r3, r4, r5]]
+fig, ax = plt.subplots(figsize=(6, 2), layout='tight')
+ax.plot(XX, rough_surface_lam(XX, r11, r22, r33, r44, r55))
+ax.set_xlabel('[m]')
+ax.set_ylabel('[m]')
+_ = ax.set_title('Road Profile')
+# %%
+# Animate the Solution
+# --------------------
+fps = 100
```

Comment:
You could mention that the fps is higher to clearly show the fast changing acceleration vectors.